Wrappers/Python/ccpi/optimisation/algs.py

```diff
@@ -42,45 +42,63 @@ def FISTA(x_init, f=None, g=None, opt=None):
     
     # algorithmic parameters
     if opt is None: 
-        opt = {'tol': 1e-4, 'iter': 1000}
-    else:
-        try:
-            max_iter = opt['iter']
-        except KeyError as ke:
-            opt[ke] = 1000
-        try:
-            opt['tol'] = 1000
-        except KeyError as ke:
-            opt[ke] = 1e-4
-    tol = opt['tol']
-    max_iter = opt['iter']
+        opt = {'tol': 1e-4, 'iter': 1000, 'memopt':False}
     
+    max_iter = opt['iter'] if 'iter' in opt.keys() else 1000
+    tol = opt['tol'] if 'tol' in opt.keys() else 1e-4
+    memopt = opt['memopts'] if 'memopts' in opt.keys() else False
+        
+        
     # initialization
-    x_old = x_init
-    y = x_init;
+    if memopt:
+        y = x_init.clone()
+        x_old = x_init.clone()
+        x = x_init.clone()
+        u = x_init.clone()
+    else:
+        x_old = x_init
+        y = x_init;
     
     timing = numpy.zeros(max_iter)
     criter = numpy.zeros(max_iter)
     
     invL = 1/f.L
     
     t_old = 1
+    
 
     # algorithm loop
     for it in range(0, max_iter):
     
         time0 = time.time()
-        
-        u = y - invL*f.grad(y)
-        
-        x = g.prox(u,invL)
-        
-        t = 0.5*(1 + numpy.sqrt(1 + 4*(t_old**2)))
-        
-        y = x + (t_old-1)/t*(x-x_old)
-        
-        x_old = x
-        t_old = t
+        if memopt:
+            # u = y - invL*f.grad(y)
+            # store the result in x_old
```

Comment:
The comment here "store the result in x_old" seems misplaced, as this normally happens at the end of the loop. In fact, it seems the code does not update x_old, which it should.